engine/crates/engine-v2/schema/src/conversion.rs

```diff
@@ -19,39 +20,46 @@ impl From<federated_graph::FederatedGraph> for Schema {
             objects: graph.objects.into_iter().map(Into::into).collect(),
             object_fields: graph.object_fields.into_iter().map(Into::into).collect(),
             fields: vec![],
-            field_types: graph.field_types.into_iter().map(Into::into).collect(),
+            types: graph.field_types.into_iter().map(Into::into).collect(),
             interfaces: graph.interfaces.into_iter().map(Into::into).collect(),
             interface_fields: graph.interface_fields.into_iter().map(Into::into).collect(),
             enums: graph.enums.into_iter().map(Into::into).collect(),
             unions: graph.unions.into_iter().map(Into::into).collect(),
             scalars: graph.scalars.into_iter().map(Into::into).collect(),
-            input_objects: graph.input_objects.into_iter().map(Into::into).collect(),
+            input_objects: vec![],
             strings: graph.strings,
             resolvers: vec![],
             definitions: vec![],
+            input_values: vec![],
         };
 
         let root_fields = {
             let mut root_fields = vec![];
-            for field_id in schema.object_fields(schema.root_operation_types.query) {
-                root_fields.push(field_id);
+            let walker = schema.default_walker();
+            for field in walker.walk(schema.root_operation_types.query).fields() {
+                root_fields.push(field.id);
             }
             if let Some(mutation) = schema.root_operation_types.mutation {
-                for field_id in schema.object_fields(mutation) {
-                    root_fields.push(field_id);
+                for field in walker.walk(mutation).fields() {
+                    root_fields.push(field.id);
                 }
             }
             if let Some(subscription) = schema.root_operation_types.subscription {
-                for field_id in schema.object_fields(subscription) {
-                    root_fields.push(field_id);
+                for field in walker.walk(subscription).fields() {
+                    root_fields.push(field.id);
                 }
             }
             root_fields.sort_unstable();
             root_fields
         };
 
         // Yeah it's ugly, conversion should be cleaned up once we got it working I guess.
-        // -- RESOLVERS --
+        // -- FIELDS & RESOLVERS --
+        // 1. The federated graph uses "resolvable_in" whenever a field is present in a subgraph.
+        //    But for resolvers we only want the "entrypoints", so root fields and later the `@key`
+        //    for federation entities.
+        // 2. Field arguments are converted to input values. That's how the GraphQL spec defines
+        //    them and having an id allows data sources to rename those more easily.
```

Comment:
yeah definitely, the closer the federated graph is the better imho. But it doesn't need to be the same either. I also wanted to separate the enum values into their separate `Vec` with `EnumValueId` which would allow us later to rename enum values (graphql/openapi connectors typically). But well, for another time, when it'll actually be needed :D

Reply:
Totally. I just think it's good to seize the chance right now while we still can relatively easily make breaking changes.